framework/wazuh/logtest.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (C) 2015-2020, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+from wazuh import WazuhError
+from wazuh.core.logtest import send_logtest_msg
+from wazuh.rbac.decorators import expose_resources
+
+
+@expose_resources(actions=['logtest:run'], resources=['*:*:*'])
+def run_logtest(token=None, event=None, log_format=None, location=None):
+    """Get the logtest output after sending a JSON to its socket.
+
+    Parameters
+    ----------
+    token : str, optional
+        Logtest session token. Default `None`
+    event : str
+        Log event.
+    log_format : str
+        Log format.
+    location : str
+        Log location.
+
+    Raises
+    ------
+    WazuhError(7000)
+        If there are more kwargs than expected.
+
+    Returns
+    -------
+    dict
+        Logtest response after analyzing the event.
+    """
+    # Token is not required
+    if locals()['token'] is None:
+        del locals()['token']
+
+    response = send_logtest_msg(command='log_processing', parameters=locals())
+    if response['error'] != 0:
+        raise WazuhError(code=7000, extra_message=response.get('message', 'Could not parse error message'))
+
+    return response
+
+
+@expose_resources(actions=['logtest:end_session'], resources=['*:*:*'])
```

Comment:
I believe we should stick with a single action for logtest. Don't see use cases for having both of them. It may create more problems than advantages.

Reply:
Done.